src/engine/sparqlExpressions/SparqlExpressionTypes.h

```diff
@@ -237,17 +237,20 @@ constexpr static qlever::ResultType expressionResultTypeToQleverResultType() {
 }
 
 /// Get Id of constant result of type T.
-template <SingleExpressionResult T, typename LocalVocab>
-Id constantExpressionResultToId(T&& result, LocalVocab& localVocab,
-                                bool isRepetitionOfConstant) {
+template <SingleExpressionResult T, typename LocalVocabT>
+Id constantExpressionResultToId(T&& result, LocalVocabT& localVocab,
+                                [[maybe_unused]] bool isRepetitionOfConstant) {
   static_assert(isConstantResult<T>);
   if constexpr (ad_utility::isSimilar<T, string>) {
-    // Return the index in the local vocabulary.
-    if (!isRepetitionOfConstant) {
-      localVocab.push_back(std::forward<T>(result));
-    }
-    return Id::makeFromLocalVocabIndex(
-        LocalVocabIndex::make(localVocab.size() - 1));
+    // TODO: Should we make this more efficient by profiting from the
+    // knowledge in `isRepetitionOfConstant`? The previous code did this (out of
+    // neccessity because it just appended words to the local vocab vector).
+    return localVocab.getIdAndAddIfNotContained(std::forward<T>(result));
+    // if (!isRepetitionOfConstant) {
+    //   localVocab.addWordNoMatterWhat(std::forward<T>(result));
+    // }
+    // return Id::makeFromLocalVocabIndex(
+    //     LocalVocabIndex::make(localVocab.size() - 1));
```

Comment:
We probably should, and the easiest way would probably be a function
`localVocab.getIndexOfLastAddedWord()`
(which is internally just `size() - 1`, but we shouldn't expose these details).